### ISSUE TITLE
Include directories with transitive dependencies

### ISSUE THREAD

**Issue by SoilRos** [dokempf/xeus-cling-cmake-setup#1]
The variable `XEUSCLING_INCLUDE_DIRECTORIES` is being populated with the configuration-time include directories of the passed targets. This is fine if the target was explicitely requested to include directores (i.e. `target_include_directories(target PUBLIC path/)`). This however, does not seem to pass transitive dependencies set on targets. To solve this, one has to use generator expressions to request include directories at generation-time.

This lines below generate the desired pragmas with one exception: new lines are written as `\n` instead as a new line. 

```cmake
foreach(target ${XEUSCLING_TARGETS})
   # Append all include directories to the pragma header
   list(APPEND XEUSCLING_INCLUDE_DIRECTORIES "$<TARGET_PROPERTY:${target},INCLUDE_DIRECTORIES>")
endforeach()

set(xeus_pragma_header "$<$<BOOL:${XEUSCLING_INCLUDE_DIRECTORIES}>:#pragma cling add_include_path(\"$<JOIN:${XEUSCLING_INCLUDE_DIRECTORIES},\")\\n #pragma cling add_include_path(\">>\")\\n")
```

**Comment by SoilRos** [comment]
This probably also happens to the other properties

**Comment by dokempf** [comment]
Thanks for bringing this up. I will try to implement this as you describe. I have not fully understood the newline problem yet, but it sounds like a problem that can be solved with even uglier generator expressions.

**Comment by dokempf** [comment]
@SoilRos I have tried this and my first attempt works on a simple problem and does not produce newline weirdnesses. Maybe you can have a look at #3

**Comment by dokempf** [comment]
I am not sure what other properties are affected. Libraries are dealt with through how shared libraries work - you do not need to specify the transitive dependencies in that case.

**Comment by SoilRos** [comment]
Other properties:
* `INTERFACE_LINK_LIBRARIES` (although I have to accept that I don't know any usage if we always work on shared libraries)
* `INTEFACE_COMPILE_OPTIONS`: Options that any file inking against the library should include.
* `INTEFACE_COMPILE_DEFINITIONS`: Same as options nut starting with `-D`.

They should be `INTERFACE` because we only need usage requirements, not build requirements.

**Comment by dokempf** [comment]
Fixed by #6
